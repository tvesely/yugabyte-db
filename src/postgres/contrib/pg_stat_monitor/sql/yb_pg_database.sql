CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;

CREATE DATABASE db1;
CREATE DATABASE db2;

\c db1
CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
CREATE TABLE t1 (a int);
CREATE TABLE t2 (b int);

\c db2
CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
CREATE TABLE t3 (c int);
CREATE TABLE t4 (d int);

SELECT pg_stat_monitor_reset();
\c db1
SELECT * FROM t1,t2 WHERE t1.a = t2.b;

\c db2
SELECT * FROM t3,t4 WHERE t3.c = t4.d;

SELECT datname, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT pg_stat_monitor_reset();

\c db1
DROP TABLE t1;
DROP TABLE t2;

\c db2
DROP TABLE t3;
DROP TABLE t4;

\c yugabyte
DROP DATABASE db1;
DROP DATABASE db2;
DROP EXTENSION IF EXISTS pg_stat_monitor;
