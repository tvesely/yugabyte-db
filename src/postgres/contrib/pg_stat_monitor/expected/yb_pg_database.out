CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
CREATE DATABASE db1;
CREATE DATABASE db2;
\c db1
CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
CREATE TABLE t1 (a int);
CREATE TABLE t2 (b int);
\c db2
CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
CREATE TABLE t3 (c int);
CREATE TABLE t4 (d int);
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

\c db1
SELECT * FROM t1,t2 WHERE t1.a = t2.b;
 a | b
---+---
(0 rows)

\c db2
SELECT * FROM t3,t4 WHERE t3.c = t4.d;
 c | d
---+---
(0 rows)

SELECT datname, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
 datname |                                 query
---------+------------------------------------------------------------------------
 db1     | SELECT * FROM t1,t2 WHERE t1.a = t2.b;
 db2     | SELECT * FROM t3,t4 WHERE t3.c = t4.d;
 db2     | SELECT datname, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
 db2     | SELECT pg_stat_monitor_reset();
(4 rows)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

\c db1
DROP TABLE t1;
DROP TABLE t2;
\c db2
DROP TABLE t3;
DROP TABLE t4;
\c yugabyte
DROP DATABASE db1;
DROP DATABASE db2;
DROP EXTENSION IF EXISTS pg_stat_monitor;
